### ISSUE TITLE
[Solved]The dxgi.dll module error in win10 causes adl.exe to crash

### ISSUE THREAD

**Issue by DJMaxIIDX** [airsdk/Adobe-Runtime-Support#418]
**AdobeTracker:**
https://tracker.adobe.com/#/view/AIR-4198924

## Problem Description
In windows10 1909 (OS Internal version 18363.959), If run the code(which is below) at the beginning of the AIR for Desktop program, the application will be crash.
```
import flash.display.StageDisplayState;
stage.displayState = StageDisplayState.FULL_SCREEN;
stage.nativeWindow.alwaysInFront = true;
```

## Steps to Reproduce
1.Create a new File with AIR for Desktop;
2.Press Ctrl+Shift+F12 to open the publish setting panel;
3.Select anly SDK from AIR27.0 to AIR33.1 and confirm;
4.Select the first frame in timeline and press F9 to open the action panel;
5.Copy & paste the code below to the action panel:
```
import flash.display.StageDisplayState;
stage.displayState = StageDisplayState.FULL_SCREEN;
stage.nativeWindow.alwaysInFront = true;
```
6.Press Ctrl+Shift+Enter to start debug;

## Actual Result:
The adl.exe crashed :(
Windows error report log will be found in Event Viewer program. Below is the sample:

> Error Application Name: adl.exe, Version: 33.1.1.190, Timestamp: 0x5f074bd7
> Error Module Name: dxgi.dll, Version: 10.0.18362.815, Timestamp: 0x390434f8
> Exception Code: 0xc0000005
> Error Offset: 0x00045f23
> Error Process ID: 0x3fb4
> Error Application Startup Time: 0x01d66c8b67e6ece9
> Error Application Path: E:\Works\SDK\AIR33.1\bin\adl.exe
> Error Module Path: C:\Windows\system32\dxgi.dll
> Report ID: 1d130c30-ae9d-4f03-b064-98a49d4232bb

## Expected Result:
Normaly, the application will fullscreen and show infront of other window.
The code above runs correctly on my other laptop with Win10 1809 version installed, the dxgi.dll file version in that system is 10.0.17763. so I think maybe this error is related to some screen draw api in DirectX 12.

## Any Workarounds:
1.Delay code execution by timer, enterframe event, or run the code after the first frame.
```
import flash.display.StageDisplayState;
import flash.utils.Timer;
import flash.events.TimerEvent;
var delayRun: Timer = new Timer(0, 1);
delayRun.addEventListener(TimerEvent.TIMER_COMPLETE, function (){
	stage.displayState = StageDisplayState.FULL_SCREEN;
	stage.nativeWindow.alwaysInFront = true;
});
delayRun.start();
```
2.Use the AIR SDK below 27.0;

**Comment by itlancer** [comment]
I have seen similar crash using `PrintJob` in fullscreen mode for some devices.

**Comment by ajwfrost** [comment]
Hi - we've been unable to reproduce this here. Using Win10 1909 and with the same version of this dll, 10.0.18362.815
Tried both AIR 32 and also 33.1.1.259.

If you can reproduce it with a 33.1.x build,, are you able to then provide a kdmp file so that we can see the call stack?

thanks

**Comment by DJMaxIIDX** [comment]
> Hi - we've been unable to reproduce this here. Using Win10 1909 and with the same version of this dll, 10.0.18362.815
> Tried both AIR 32 and also 33.1.1.259.
> 
> If you can reproduce it with a 33.1.x build,, are you able to then provide a kdmp file so that we can see the call stack?
> 
> thanks

Maybe the issue can olny reproduce in 1909.18363.x, and your system is 1909.18362.x :D

I uploaded the dmp file（test with AIR33.1.1.259）:
[https://www.dropbox.com/s/bj5ry6zpxagd4mi/crash.dmp?dl=0](url)
Hope for help, thanks very much.

**Comment by DJMaxIIDX** [comment]
I tested with SDK 33.1.1.300 in Win10 2004 (19041.630)，the issue still exists.

**Comment by piii866** [comment]
the same problem.

19.0sdk  is work fine

@DJMaxIIDX ,can u share 27.0 sdk，

**Comment by DJMaxIIDX** [comment]
> the same problem.
> 
> 19.0sdk is work fine
> 
> @DJMaxIIDX ,can u share 27.0 sdk，thank you

https://www.dropbox.com/s/hqviragyj1phzoh/AIRSDK_27.0.0.132.zip?dl=0

**Comment by ajwfrost** [comment]
FYI one of the guys took a look at this and has managed to reproduce it, it sounds like it's a D3D11 issue but we can work around it (i.e. it's the sort of thing where they should return a failure code rather than crashing...)

**Comment by DJMaxIIDX** [comment]
The issue has been fixed in v33.1.1.533, thank you!!
